rfc/graphql-schema/additionalTypeDefs/byPage/userManagement.ts

```diff
@@ -0,0 +1,108 @@
+export default /* GraphQL */ `
+  input CustomerFiltersInput {
+    search: String
+    companyRoleId: ID
+  }
+
+  type Role {
+    id: ID!
+    name: String!
+  }
+
+  extend type Customer {
+    role: Role
+  }
+
+  type CustomerEdge {
+    node: Customer!
+    cursor: String!
+  }
+
+  type CustomerConnection {
+    edges: [CustomerEdge!]
+    pageInfo: PageInfo!
+  }
+
+  type Company implements Node {
+    id: ID!
+    Customers(
+      filters: CustomerFiltersInput
+      before: String
+      after: String
+      first: Int
+      last: Int
+    ): CustomerConnection!
+  }
+
+  # Placeholder for real domain errors
+  type SomeChangeCustomerRoleError implements Error {
+    message: String!
+  }
+
+  # Placeholder for real domain errors
+  type AnotherChangeCustomerRoleError implements Error {
+    message: String!
+  }
+
+  union ChangeCustomerRoleError =
+      SomeChangeCustomerRoleError
+    | AnotherChangeCustomerRoleError
+
+  type ChangeCustomerRoleResult {
+    errors: [ChangeCustomerRoleError!]!
+    customer: Customer
+  }
+
+  # Placeholder for real domain errors
+  type SomeDeregistrationError implements Error {
+    message: String!
+  }
+
+  # Placeholder for real domain errors
+  type AnotherDeregistrationError implements Error {
+    message: String!
+  }
+
+  union DeregisterCustomerError =
+      SomeDeregistrationError
+    | AnotherDeregistrationError
+
+  type DeregistrationResult {
+    errors: [DeregisterCustomerError!]!
+  }
+
+  # Placeholder for real domain errors
+  type SomeDeregistrationError implements Error {
+    message: String!
+  }
+
+  # Placeholder for real domain errors
+  type AnotherDeregistrationError implements Error {
+    message: String!
+  }
+
+  input RegisterCompanyCustomerInput {
+    firstName: String!
+    lastName: String!
+    email: String!
+    roleId: ID!
+    phone: String
+    formFields: CustomerFormFieldsInput
+  }
+
+  type CompanyMutations {
+    changeCustomerRole(customerId: Int!, roleId: ID!): ChangeCustomerRoleResult!
+    deregisterCustomer(customerId: Int!): DeregistrationResult!
```

Comment:
we might need to flag this sooner rather than later since it is a change in behaviour.

Reply:
in theory it could act the same way (deleting)

Reply:
I wonder if deregistering gets rid of the customer account, or just disconnects the customer from the company?